src/api/routes/auth/register.ts

```diff
@@ -287,6 +287,16 @@ router.post(
 			});
 		}
 
+		const { maxUsername } = Config.get().limits.user;
+		if (body.username.length > maxUsername) {
+			throw FieldErrors({
+				username: {
+					code: "USERNAME_INVALID",
```

Comment:
are you sure this is the right code, and not `BASE_TYPE_BAD_LENGTH`?

Reply:
I'm not, I've just used the already existing check from `PATCH /api/users/@me`:
https://github.com/spacebarchat/server/blob/bb31df036a8067681729623f7c41dedd6b3a94e9/src/api/routes/users/%40me/index.ts#L155-L159

Reply:
I checked how discord does it and it returns
```{
    "message": "Invalid Form Body",
    "code": 50035,
    "errors": {
        "username": {
            "_errors": [
                {
                    "code": "BASE_TYPE_BAD_LENGTH",
                    "message": "Must be between 2 and 32 in length."
                }
            ]
        }
    }
}```

for the changing the username, so ig both should be changed